core/io/marshalls.cpp

```diff
@@ -657,6 +672,21 @@ Error decode_variant(Variant &r_variant, const uint8_t *p_buffer, int p_len, int
 							return err;
 						}
 
+						if (str == "script") {
+							String path;
+							err = _decode_string(buf, len, r_len, path);
```

Comment:
I think that even if it didn't work properly before (creating a new script instead of loading an existing one, causing the type checks to fail), we should still use a new flag (`HEADER_DATA_OBJECT_SCRIPT_AS_PATH`), for compatibility reasons?

Reply:
I used `decode_variant()` instead of `_decode_string()` and added a type check, now it doesn't break backwards compatibility.